client/purplecaffeine/core.py

```diff
@@ -0,0 +1,267 @@
+        """Adds metric to trial data.
+
+        Args:
+            name: name of metric
+            value: value of metric
+        """
+        self.metrics.append([name, value])
+
+    def add_parameter(self, name: str, value: str):
+        """Adds parameter to trial data.
+
+        Args:
+            name: name of the parameter, like OS
+            value: value for the parameter, like Ubuntu
+        """
+        self.parameters.append([name, value])
+
+    def add_circuit(self, name: str, circuit: QuantumCircuit):
+        """Adds circuit to trial data.
+
+        Args:
+            name: name of the circuit
+            circuit: QuantumCircuit
+        """
+        self.circuits.append([name, circuit])
+
+    def add_operator(self, name: str, operator: Operator):
+        """Adds operator to trial data.
+
+        Args:
+            name: name of the parameter
+            operator: quantum Operator
+        """
+        self.operators.append([name, operator])
+
+    def add_artifact(self, name: str, artifact: Any):
+        """Adds artifacts path to trial data.
+
+        Args:
+            name: name of the file
+            artifact: file object
+        """
+        if asizeof.asizeof(artifact) >= Configuration.MAX_SIZE:
+            logging.warning(
+                "Your file is too big ! Limit : %s Bytes", str(Configuration.MAX_SIZE)
+            )
+        self.artifacts.append([name, artifact])
+
+    def add_text(self, title: str, text: str):
+        """Adds any text to trial data.
+
+        Args:
+            title: title of the text
+            text: long string
+        """
+        self.texts.append([title, text])
+
+    def add_array(self, name: str, array: Union[np.ndarray, List[Any]]):
+        """Adds array to trial data.
+
+        Args:
+            name: name of the array
+            array: quantum circuit results
+        """
+        self.arrays.append([name, array])
+
+    def add_tag(self, tag: str):
+        """Adds any tag to trial data.
+
+        Args:
+            tag: word of your tag
+        """
+        self.tags.append(tag)
+
+    def save(self):
+        """Save into Backend."""
+        self.backend.save(name=self.name, trial=self)
+
+    def read_trial(self) -> "Trial":
+        """Read a trial from Backend.
+
+        Returns:
+            Trial dict object
+        """
+        return self.backend.get(name=self.name)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.save()
+
+
+class BaseBackend:
+    """Base backend class."""
+
+    def save(self, name: str, trial):
+        """Saves given trial.
+
+        Args:
+            name: name of the trial
+            trial: encode trial to save
+        """
+        raise NotImplementedError
+
+    def list(
+        self,
+        query: Optional[str] = None,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        **kwargs,
+    ) -> List["Trial"]:
```

Comment:
```suggestion
    ) -> List[Trial]:
```

Reply:
it is the same thing. It's syntactic sugar to avoid accessing classes that are not defined yet